### DISCUSSION TITLE
Changing an object color in a custom shape based on output.

### DISCUSSION THREAD

**Discussion by yousefbinali** [hneemann/Digital#1018] [Q&A] [answered]
I want to make a custom display that has x segments. I made a circuit that has x outputs. I want to make custom shape for this circuit, and change the color of x segments based on the output pins. 
How can I accomplish this if it is possible?

**Comment by hneemann** [comment, marked as answer]
This is only possible by implementing such a display in Java.
See this [example](https://github.com/hneemann/digitalCustomComponents) on how a custom component is implemented.

**Reply by yousefbinali** [reply]
Thank you.
